Packs/Traceable/Integrations/Traceable/Traceable.py

```diff
@@ -616,41 +686,111 @@ def get_threat_events(
+        span_id_map = {}
+        demisto.info("Extracting spans from threads.")
+        for future in future_list:
+            trace_results = future.result()
+            if Helper.is_error(trace_results, "data", "spans", "results"):
+                msg = f"Error Object: {json.dumps(trace_results)}. Couldn't get the Span."
+                demisto.info(msg)
+            else:
+                if len(trace_results["data"]["spans"]["results"]) > 0:
+                    traces = trace_results["data"]["spans"]["results"]
+                    for trace in traces:
+                        if ("id" in trace
+                            and trace["id"] is not None
+                            and trace["id"] != ""
+                                and trace["id"] not in span_id_map):
+                            span_id_map[trace["id"]] = trace
+                else:
+                    demisto.info("Didn't find any spans. Span array length:"
+                                 + f" {len(trace_results['data']['spans']['results'])}."
+                                 + f" Span Object: {json.dumps(trace_results['data']['spans']['results'])}")
 
         api_endpoint_details = []
         api_endpoint_details_map = {}
 
         if self.__query_api_attributes:
             api_endpoint_details = self.get_api_endpoint_details(
-                api_id_list, starttime, endtime
+                list(api_id_map.keys()), starttime, endtime
             )
             api_endpoint_details_map = {
                 api_endpoint_detail["id"]: api_endpoint_detail
                 for api_endpoint_detail in api_endpoint_details
             }
         api_endpoint_details = None
 
-        for domain_event, future in future_list:
-            trace_results = future.result()
+        for domain_event in results:
+            if (
+                "spanId" in domain_event
+                    and "value" in domain_event["spanId"]
+                    and domain_event["spanId"]["value"] is not None
+                    and domain_event["spanId"]["value"] != ""
+                    and domain_event["spanId"]["value"] in span_id_map
+            ):
+                domain_event["spans"] = span_id_map[domain_event["spanId"]["value"]]
+            demisto.info("Done waiting for the future object...")
             domain_event["type"] = "Exploit"
+            if (
+                domain_event["environment"] is not None
+                and domain_event["environment"]["value"] is not None
+            ):
+                domain_event["environment"] = domain_event["environment"]["value"]
+            if (
+                domain_event["serviceName"] is not None
+                and domain_event["serviceName"]["value"] is not None
+            ):
+                domain_event["serviceName"] = domain_event["serviceName"]["value"]
+            if (
+                domain_event["apiId"] is not None
+                and domain_event["apiId"]["value"] is not None
+            ):
```

Comment:
You can use the dict `get` method here and everywhere else
https://www.w3schools.com/python/ref_dictionary_get.asp